File: src/main/java/io/saso/dash/services/SubServiceFactory.java
package io.saso.dash.services;

import com.google.inject.name.Named;
import org.kohsuke.github.GitHub;

public interface SubServiceFactory
{
    @Named("github-notification")
    Pollable createGitHubNotificationSubService(Service superService,
                                                GitHub gitHub);
}

File: src/main/kotlin/io/saso/dash/services/SubServiceFactory.kt
package io.saso.dash.services

import com.google.inject.name.Named
import org.kohsuke.github.GitHub

public interface SubServiceFactory
{
    @Named("github-notification")
    fun createGitHubNotificationSubService(
            superService: Service, gitHub: GitHub): ServicePollable
}
